packages/google_maps_flutter/example/test_driver/google_maps_e2e.dart

```diff
@@ -376,6 +376,35 @@ void main() {
     expect(secondVisibleRegion.contains(newCenter), isTrue);
   });
 
+  testWidgets('testMyLocation', () async {
+    final Key key = GlobalKey();
+    final Completer<GoogleMapInspector> inspectorCompleter =
+        Completer<GoogleMapInspector>();
+
+    await pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: GoogleMap(
+        key: key,
+        initialCameraPosition: _kInitialCameraPosition,
+        myLocationEnabled: true,
+        myLocationButtonEnabled: false,
+        onMapCreated: (GoogleMapController controller) {
+          final GoogleMapInspector inspector =
+              // ignore: invalid_use_of_visible_for_testing_member
+              GoogleMapInspector(controller.channel);
+          inspectorCompleter.complete(inspector);
+        },
+      ),
+    ));
+
+    final GoogleMapInspector inspector = await inspectorCompleter.future;
+    final bool myLocationEnabled = await inspector.isMyLocationEnabled();
+    expect(myLocationEnabled, true);
```

Comment:
We should pump another frame with myLocationEnabled set to false and assert again. (a faulty isMyLocationEnabled implementation that always returns true will pass this test)

Reply:
We can't test it. Because this params use permissions. 

```java 
@SuppressLint("MissingPermission")
  private void updateMyLocationSettings() {
    if (hasLocationPermission()) {
      // The plugin doesn't add the location permission by default so that apps that don't need
      // the feature won't require the permission.
      // Gradle is doing a static check for missing permission and in some configurations will
      // fail the build if the permission is missing. The following disables the Gradle lint.
      //noinspection ResourceType
      googleMap.setMyLocationEnabled(myLocationEnabled);
      googleMap.getUiSettings().setMyLocationButtonEnabled(myLocationButtonEnabled);
    } else {
      // TODO(amirh): Make the options update fail.
      // https://github.com/flutter/flutter/issues/24327
      Log.e(TAG, "Cannot enable MyLocation layer as location permissions are not granted");
    }
  }
```

We should remove `skip' when https://github.com/flutter/flutter/issues/12561 will be fixed